code/game/atoms_health.dm

```diff
@@ -0,0 +1,293 @@
+ * Has no pre-modification checks.
+ * Returns `TRUE` if the death state changes, `null` if the atom is not using health, `FALSE` otherwise.
+ */
+/atom/proc/set_health(new_health)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	var/health_mod = new_health - health_current
+	return mod_health(health_mod)
+
+/**
+ * Restore's the atom's health by the given value. Returns `TRUE` if the restoration resulted in a death state change.
+ */
+/atom/proc/restore_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return mod_health(damage, damage_type = null)
+
+/**
+ * Damage's the atom's health by the given value. Returns `TRUE` if the damage resulted in a death state change.
+ * Resistance and weakness modifiers are applied here.
+ */
+/atom/proc/damage_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+
+	// Apply resistance/weakness modifiers
+	damage *= get_damage_resistance(damage_type)
+
+	return mod_health(-damage, damage_type = null)
+
+/**
+ * Proc called after any health changes made by the system
+ */
+/atom/proc/post_health_change(health_mod, damage_type = null)
+	return
+
+/**
+ * Forces the atom to a state considered alive or dead. Should be overridden to make any necessary changes to ensure the atom's state is correct.
+ * Does not allow calling parent to avoid proc chain call ordering issues with `handle_death_change()` - Overrides should include their own calls to this proc once all necessary changes are complete.
+ */
+/atom/proc/set_death_state(new_death_state)
+	SHOULD_CALL_PARENT(FALSE)
+	if (!has_health())
+		return
+	if (new_death_state != is_alive())
+		return FALSE
+	health_current = new_death_state ? 0 : get_max_health()
+	handle_death_change(new_death_state)
+	return TRUE
+
+/**
+ * Ease of use proc for `set_death_state()`. This one kills the atom.
+ */
+/atom/proc/kill_health()
+	SHOULD_CALL_PARENT(TRUE)
+	return set_death_state(TRUE)
+
+/**
+ * Ease of use proc for `set_death_state()`. This one revives the atom.
+ */
+/atom/proc/revive_health()
+	SHOULD_CALL_PARENT(TRUE)
+	return set_death_state(FALSE)
+
+/**
+ * Proc called when the death state changes changes.
+ * Provided for override use.
+ */
+/atom/proc/handle_death_change(new_death_state)
+	return
+
+/**
+ * Sets the atoms maximum health to the new value.
+ * If `set_current_health` is `TRUE`, also sets the current health to the new value.
+ */
+/atom/proc/set_max_health(new_max_health, set_current_health = TRUE)
+	SHOULD_CALL_PARENT(TRUE)
+	health_max = new_max_health
+	health_current = set_current_health ? get_max_health() : min(health_current, get_max_health())
+
+/**
+ * Sets the atom's resistance/weakness to the given damage type.
+ * Value should be a multiplier that is applied against damage. Values below 1 are a resistance, above 1 are a weakness.
+ * Value of `0` is considered immunity.
+ */
+/atom/proc/set_damage_resistance(damage_type, resistance_value)
```

Comment:
As far as I can tell this proc and the associated `health_resistances` var is only ever used by blobs. Putting it at atom level means things like areas are going to inherit it. It's not much overhead as it is LAZY but I'm not sure how ideal it is for it to be at the base object type level when only one thing in the game uses it.

Reply:
Plans are for everything that has any form of health (With the exception of human mobs themselves because of complicated organ code) to use the new health stuff, and a number of things across the code base do have some form of armor/resistance based values in their current implementations that'll have to be transferred over. Walls specifically come to mind as an example, as I've already tied those in on Bay code. I believe some subtypes of structures and simple mobs have resistance values as well.